optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/shift/view/ShiftView.java

```diff
@@ -39,29 +47,31 @@
     @SuppressWarnings("unused")
     public ShiftView() {}
 
-    public ShiftView(Integer tenantId, Spot spot, TimeSlot timeSlot) {
-        this(tenantId, spot, timeSlot, null);
+    public ShiftView(Integer tenantId, Spot spot, OffsetDateTime startDateTime, OffsetDateTime endDateTime) {
+        this(tenantId, spot, startDateTime, endDateTime, null);
     }
 
-    public ShiftView(Integer tenantId, Spot spot, TimeSlot timeSlot, Employee rotationEmployee) {
+    public ShiftView(Integer tenantId, Spot spot, OffsetDateTime startDateTime, OffsetDateTime endDateTime, Employee rotationEmployee) {
         super(tenantId);
         this.spotId = spot.getId();
-        this.timeSlotId = timeSlot.getId();
+        this.startDateTime = startDateTime;
+        this.endDateTime = endDateTime;
         this.rotationEmployeeId = (rotationEmployee == null) ? null : rotationEmployee.getId();
     }
 
     public ShiftView(Shift shift) {
         super(shift);
         this.spotId = shift.getSpot().getId();
-        this.timeSlotId = shift.getTimeSlot().getId();
-        this.lockedByUser = shift.isLockedByUser();
+        this.startDateTime = shift.getStartDateTime();
+        this.endDateTime = shift.getEndDateTime();
+        this.lockedByUser = shift.isPinnedByUser();
```

Comment:
this will need to change to pinnedByUser, seperate jira/PR

Reply:
https://issues.jboss.org/browse/PLANNER-1032